### ISSUE TITLE
Could the tags be added as "automatic" tags?

### ISSUE THREAD

**Issue by retorquere** [bwiernik/zotero-shortdoi#16]
Zotero distinguishes between automatic (with a `type` field set to `1`) and user-entered tags. If the DOI manager tags could be added as `automatic` tags, I could filter them out for export, as they're more for reference management than descriptive of the reference itself.

**Comment by bwiernik** [comment]
I'm not sure. @dstillman is that possible?

Otherwise, you could filter them using the preferences: `extensions.shortdoi.tag_invalid`, `extensions.shortdoi.tag_multiple`, and `extensions.shortdoi.tag_nodoi`

**Comment by dstillman** [comment]
You can certainly create automatic tags. I'm not sure if it's appropriate here.

FWIW, in the past we've thought about adding [another tag type](https://github.com/zotero/zotero/issues/1197) for private/workflow tags, which wouldn't show up in various public contexts. I don't know if it's worth the additional complexity, but it's something we'd be able to do fairly soon, after the current beta code lands.

**Comment by bwiernik** [comment]
Do I do that by adding an argument to `setTag()`?

There were some things had I thought about doing with Zotfile a few years back that would have really benefitted from private/workflow tags. I think they would be great.

In this case, these tags are things the user might want to filter on to resolve, would "private" tags be visible in the tag viewer?

**Comment by dstillman** [comment]
```javascript
/**
 * Set the item's tags
 *
 * A separate save() is required to update the database.
 *
 * @param {String[]|Object[]} tags - Array of strings or object in API JSON format
 *                                   (e.g., [{tag: 'tag', type: 1}])
 */
Zotero.Item.prototype.setTags = function (tags) {
```

`type: 1` means automatic. `type: 0` or omitted means manual.

> would "private" tags be visible in the tag viewer?

I think you'd be able to toggle different tag types off and on in the tag selector menu. Probably the same for exports and reports. We'd have to think about other stuff — e.g., hiding them from public (non-member) views of a library.

**Comment by retorquere** [comment]
@bwiernik I can't read those preferences in a translator.

**Comment by stefanct** [comment]
Alternatively (or even additionally) an option to disable the tagging altogether would be fine (with me) too.

Is it correct that simply removing the tag strings for the various reasons in the settings is the correct way to make it work that way? It looks like it is working but I'd like to avoid breakage in the future because of that ;) An extra checkbox would be preferable in any case IMHO (it could preserve the original strings and allow for easy toggling).

**Comment by bwiernik** [comment]
@dstillman Is implementing private tags something that might be possible soon?

**Comment by bwiernik** [comment]
@stefanct Yes, you can disable the tags just be deleting the strings in the preferences.

**Comment by bwiernik** [comment]
Closed by 9b582d1953e505423d85ebfacc8ddc81b06c0e6b
